cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py

```diff
@@ -1033,29 +1034,411 @@ def test_aws_api_errors(self, client, mocker):
+                id="request with invalid name",
+            ),
+            pytest.param(
+                {"clusterConfiguration": BASE64_ENCODED_CONFIG},
+                "us-east-1",
+                "clusterName",
+                ["ALL", "ALLL"],
+                None,
+                None,
+                None,
+                {"message": "Bad Request: 'ALLL' does not match '^(ALL|type:[A-Za-z0-9]+)$'"},
+                id="request with invalid validator",
+            ),
+            pytest.param(
+                {"clusterConfiguration": BASE64_ENCODED_CONFIG},
+                "us-east-1",
+                "clusterName",
+                ["type:"],
+                None,
+                None,
+                None,
+                {"message": "Bad Request: 'type:' does not match '^(ALL|type:[A-Za-z0-9]+)$'"},
+                id="request with empty named validator",
+            ),
+            pytest.param(
+                {"clusterConfiguration": BASE64_ENCODED_CONFIG},
+                "us-east-1",
+                "clusterName",
+                None,
+                "CRITICAL",
+                None,
+                None,
+                {"message": "Bad Request: 'CRITICAL' is not one of ['INFO', 'WARNING', 'ERROR']"},
+                id="request with invalid validation failure level",
+            ),
+            pytest.param(
+                {"clusterConfiguration": BASE64_ENCODED_CONFIG},
+                "us-east-1",
+                "clusterName",
+                None,
+                None,
+                "NO",
+                None,
+                {"message": "Bad Request: Wrong type, expected 'boolean' for query parameter 'dryrun'"},
+                id="request with invalid dryrun value",
+            ),
+            pytest.param(
+                {"clusterConfiguration": BASE64_ENCODED_CONFIG},
+                "us-east-1",
+                "clusterName",
+                None,
+                None,
+                None,
+                "NO",
+                {"message": "Bad Request: Wrong type, expected 'boolean' for query parameter 'forceUpdate'"},
+                id="request with invalid forceUpdate value",
+            ),
+            pytest.param(
+                {"clusterConfiguration": "config"},
+                "us-east-1",
+                "clusterName",
+                None,
+                None,
+                None,
+                None,
+                {"message": "Bad Request: invalid configuration. " "Please make sure the string is base64 encoded."},
+                id="request with configuration not encoded",
+            ),
+            pytest.param(
+                {"clusterConfiguration": "aW52YWxpZA=="},
+                "us-east-1",
+                "clusterName",
+                None,
+                None,
+                None,
+                None,
+                {"message": "Bad Request: Cluster update failed.\nconfiguration must be a valid YAML document"},
```

Comment:
configuration not capitalized

Reply:
Changed old `Cluster` exception code to `f"Cluster update failed.\n{str(e).upper()}"`